ovs/lib/alba.py

```diff
@@ -1411,70 +1412,103 @@ def _remove_service(_node, _service_name, _abackend):
                 available_node_map[alba_backend.guid] = sorted(available_node_map[alba_backend.guid],
                                                                key=lambda n: node_load.get(n, 0))
 
+            layout_key = AlbaController.AGENTS_LAYOUT_CONFIG_KEY.format(alba_backend.guid)
+            layout = None
+            if Configuration.exists(layout_key):
+                layout = Configuration.get(layout_key)
```

Comment:
If layout is a list, i'd perform a check whether the node IDs actually exist, because if none of them exist, you'll end up with 0 maintenance agents for that backend